Map/Test/MapTest.cpp

```diff
@@ -32,10 +32,74 @@ TEST_F(SquareMapTest, quantityOfFieldsInFirstColumnShouldEqualMapSize)
     ASSERT_EQ(map.getFields().at(0).size(), mapSize);
 }
 
-TEST_F(SquareMapTest, fieldShouldBeEmpty)
+TEST_F(SquareMapTest, MapWithSize2ShouldContain1TreasureField)
 {
-    const auto& field = map.getFields().at(0).at(0);
-    ASSERT_EQ(field->getType(), FieldType::Empty);
+    SquareMap map(2);
+    constexpr int expectedNumberOfTreasureFields = 1;
+
+    auto &list = map.getFields();
+
+    int numberOfTreasureFields = 0;
+
+    for(unsigned i = 0; i < list.size(); i++)
+    {
+        numberOfTreasureFields += std::count_if(list.at(i).begin(), list.at(i).end(),
+            [&](const auto &field){ return field -> getType() == FieldType::Treasure; });
+    }
```

Comment:
Tu można użyć range-based for. Tak samo niżej

Reply:
słuszna uwaga ! Zmiencie na range loop ;)